packages/vhd-lib/src/Vhd/VhdAbstract.js

```diff
@@ -0,0 +1,114 @@
+import { computeBatSize, sectorsRoundUpNoZero, sectorsToBytes } from './_utils'
+import { SECTOR_SIZE } from '../_constants'
+export class VhdAbstract {
+  bitmapSize
+
+  blockTable
+  #header
+  footer
+
+  fullBlockSize
+  sectorsOfBitmap
+  sectorsPerBlock
+
+  get header() {
+    return this.#header
+  }
+
+  set header(header) {
+    this.#header = header
+    this.sectorsPerBlock = this.header.blockSize / SECTOR_SIZE
+    this.sectorsOfBitmap = sectorsRoundUpNoZero(this.sectorsPerBlock >> 3)
+    this.fullBlockSize = this.sectorsOfBitmap + this.sectorsPerBlock
+    this.bitmapSize = sectorsToBytes(this.sectorsOfBitmap)
+  }
+
+  /**
+   * instantiate a Vhd
+   *
+   * @returns {AbstractVhd}
+   */
+  static async open(opts = {}) {
+    throw new Error('open not implemented')
+  }
+
+  /**
+   * Check if this vhd contains a block with id blockId
+   * Must be called after readBlockAllocationTable
+   *
+   * @param {Number} blockId
+   * @returns {Boolean}
```

Comment:
AFAICT, `Number` and `Boolean` are incorrect and should be lowercased.

See https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#param-and-returns